### PR TITLE
dbName follows dependencies.indexedDB

### PR DESCRIPTION
**Pull request by mcshaz**
setting Dexie.dependencies.indexedDB = indexedDB;
now updates the databaseNamesEnumerator so that the table of names is in the same database provider

addresses point 1 of issue #816

still has not addressed including the IDBFactory when instantiating a new Dexie object 

all tests passing

### CODE REVIEW

**Comment by dfahlander** [comment]
Thanks for this PR! I see that tests does not currently pass. Just for your info, I'm currently having issues with travis/karms/browserstack also for the master branch but haven't nailed down the issue yet. Probably due to external causes (changes in the travis or browserstack services).

I will keep trying to find the reason for the travis issues.

**Comment by mcshaz** [comment]
thank you - it was passing until I did something - i'll try and nail it down

**Comment by dfahlander** [comment]
Just want to say, it might be problems with travis/browserstack, since I'm having issues in master branch as well, and that was before your PR.

**Comment by mcshaz** [comment]
can you please let me know when the travis/browserstack issue is fixed, and I'll have another look - thank you.

**Comment by dfahlander** [comment]
Seems the pure dexie tests pass, so that's good. Probably it's only the travis-issue that makes the addon-tests fail. I'll let you know when the travis issues are fixed, but I can't promise when that will be, but as long as it's not fixed, we're blocked on any PRs.

**Comment by dfahlander** [comment]
@mcshaz you can also subscribe to #821 to get notified about the travis-issue.

File: src/classes/dexie/dexie-static-props.ts [hunk 1]
```diff
@@ -198,17 +199,10 @@ props(Dexie, {
   // In node.js, however, these properties must be set "manually" before instansiating a new Dexie().
   // For node.js, you need to require indexeddb-js or similar and then set these deps.
   //
-  dependencies: (()=>{
-    try {
```
Comment (dfahlander) [inline src/classes/dexie/dexie-static-props.ts]: The try...catch is important here. We had an issue where Dexie fails completely to initialize when used in private mode. Better if it fails in a controlled way (Dexie still defined but fails in db.open())

**Comment by mcshaz** [comment]
this is just being closed and reopened to rerun the Travis CLI build now that issue is fixed

**Comment by mcshaz** [comment]
passing, and passes when I run tests in IE. Bizarrely, when I first ran the tests in IE, it failed for a single test with:
```
    × modify-with-where(issue-594)
      IE 11.0.0 (Windows 7.0.0)
    There should be 3 Davids
    Expected: 3
    Actual: 2
 ```
but has passed all tests on subsequent occasions. of note the original issue seems to be safari, and I am not sure if the test is truly encapsulated.

to be clear on what this test allows:
```typescript
export class DrugsDBLocal extends Dexie implements IDrugDB {
    constructor(indexedDb?: IDBFactory,
                      dbKeyRange?: typeof IDBKeyRange) {
        if (indexedDb !== void 0) {
            Dexie.dependencies.indexedDB = indexedDb as IDBFactory;
        }
        if (dbKeyRange !== void 0) {
            Dexie.dependencies.IDBKeyRange = dbKeyRange;
        }
        super('DrugData')
```

**Comment by dfahlander** [comment]
Good that the tests succeeds. Internet Explorer is shaky. The browser fail sporadically on the unit tests and always has.

What I found in the review was the try..catch being removed. The reason for the try..catch can be found in issue #614, #617 and PR #619.

**Comment by mcshaz** [comment]
No worries – I’ll put it back in.

**Comment by mcshaz** [comment]
the try catch is back in the original state. having said that, I think it could be potentially improved further - I am as far from an expert on indexedDB as could be, but the error messages & stack trace I have seen are quite unambiguous in Firefox at least - the error is because of insufficient DB permissions. I am not as convinced a null reference would give just as much information. If you want to wait until calling db open, what I would propose is something along the lines of:

```typescript
    public constructor(args?: IDexieDependencies){
        if (args !== void 0){
            this.IDBKeyRange = args.IDBKeyRange;
            this.indexedDB = args.indexedDB;
            this.pIDBError = args.iDBError;
        }
    }
    public get indexedDB(){
        if (this.pIDBError) { throw this.pIDBError}
        return this.pIndexedDB;
    }
... 'dexie-static-props
  dependencies: new DexieDependencies((() => {
    try {
      return {
        indexedDB: (_global.indexedDB || _global.mozIndexedDB || _global.webkitIndexedDB || _global.msIndexedDB) as IDBFactory,
        IDBKeyRange: (_global.IDBKeyRange || _global.webkitIDBKeyRange) as IDBKeyRange
      }
    } catch (err) {
      return { indexedDB: null, IDBKeyRange: null, iDBError: err }
    }
  })()),
```

I have not instituted this as this would be  a breaking change

**Comment by mcshaz** [comment]
just bumping this as so far no response since making the changes a month ago, and wondering if it has been forgotten  - as per your comment, I had added back in the enclosed function & try/catch block to the PR, + added a new comment explaining why the try/catch block. 

On a separate branch (not in this PR) I hold onto the original browser error in order to rethrow when first accessing Dexie

**Comment by dfahlander** [comment]
I haven't got any time for getting a full understanding of the changes. I've had a lot to do, and this PR comes with new logic that is not obvious so i would need to spend more time on it.

**Comment by mcshaz** [comment]
No worries, I had just thought it might have been forgotten about.

**Comment by dfahlander** [comment]
Replaced by #1194

**Pull request closed without merging**